episodes/basic-gis-with-r-sf.Rmd

```diff
@@ -0,0 +1,245 @@
+encode spatial vector data."](https://cran.r-project.org/web/packages/sf/sf.pdf).
+It contains a large set of functions to achieve all the operations on vector spatial data for which you might use traditional GIS software: change the coordinate system, join layers, intersect or unite polygons, create buffers and centroids, etc. cf. the `sf` [cheatsheet](https://raw.githubusercontent.com/rstudio/cheatsheets/main/sf.pdf).
+
+
+### Conservation in Brielle, NL
+
+Let's focus on old buildings and imagine we're in charge of their conservation. We want to know how much of the city would be affected by a non-construction zone of 100m around pre-1800 buildings.
+
+Let's select them and see where they are.
+
+
+```{r}
+old <- 1800 # year prior to which you consider a building old
+
+assign("has_internet_via_proxy", TRUE, environment(curl::has_internet))
+nominatim_polygon <- nominatimlite::geo_lite_sf(address = "Brielle, NL", points_only = FALSE)
+bb <- sf::st_bbox(nominatim_polygon)
+x <- opq(bbox = bb) %>%
+   add_osm_feature(key = 'building') %>%
+    osmdata_sf()
+buildings <- x$osm_polygons %>%
+  st_transform(.,crs=28992)
+
+
+summary(buildings$start_date)
+buildings$start_date <- as.numeric(as.character(buildings$start_date))
+
+old_buildings <- buildings %>%
+  filter(start_date <= old)
+
+ ggplot(data = old_buildings) + geom_sf(colour="red")
+ 
+```
+As conservationists, we want to create a zone around historical buildings where building regulation will have special restrictions to preserve historical buildings. 
+
+## Buffers
+
+Let's say this zone should be 100 meters. In GIS terms, we want to create a buffer around polygons. The corresponding function `sf` is `st_buffer`, with 2 arguments: the polygons around which to create buffers, and the radius of the buffer.
+
+
+```{r}
+ distance <- 100 # in meters 
+ 
+buffer_old_buildings <- 
+  st_buffer(x = old_buildings, dist = distance)
+ 
+ ggplot(data = buffer_old_buildings) + geom_sf()
+  
+```
+
+## Union
+
+Now, we have a lot of overlapping buffers. We would rather create a unique conservation zone rather than overlapping ones in that case. So we have to fuse the overlapping buffers into one polygon. This operation is called union and the corresponding function is `st_union`.
+
+```{r}
+ single_old_buffer <- st_union(buffer_old_buildings) %>%
+   st_cast(to = "POLYGON") %>%
+  st_as_sf() 
+
+single_old_buffer<- single_old_buffer %>%
+  mutate("ID"=as.factor(1:nrow(single_old_buffer))) %>%
+  st_transform(.,crs=28992) 
+  
+```
+
+We also use `st_cast()` to explicit the type of the resulting object (*POLYGON* instead of the default *MULTIPOLYGON*), `st_as_sf()` to transform the polygon into an `sf` object.
+
+
+## Centroids
+For the sake of visualisation speed, we would like to represent buildings by a single point (for instance: their geometric centre) rather than their actual footprint. This operation means defining their centroid and the corresponding function is `st_centroid`.
```

Comment:
```suggestion
For the sake of visualisation speed, we would like to represent buildings by a single point (for instance: their geometric centre) rather than their actual footprint. This operation means defining their _centroid_ and the corresponding function is `st_centroid()`.
```